lib/bundlex/native.ex

```diff
@@ -118,37 +157,52 @@ defmodule Bundlex.Native do
     end
   end
 
-  defp get_native_configs(project, types \\ [:lib, :nif, :cnode, :port]) do
+  defp get_native_configs(project, types \\ [:lib, :nif, :cnode, :native, :port]) do
     types
     |> Bunch.listify()
     |> Enum.flat_map(fn type ->
       project.config
       |> Keyword.get(@native_type_keys[type], [])
       |> Enum.map(fn {name, config} ->
-        %{config: config, name: name, type: type, app: project.app}
+        case type do
+          :cnode ->
+            IO.warn(":cnodes are deprecated. Use natives instead.")
+            config = Keyword.put(config, :interfaces, [:cnode])
+            %{config: config, name: name, type: type, app: project.app}
+
+          :nif ->
+            IO.warn(":nifs are deprecated. Use natives instead.")
+            config = Keyword.put(config, :interfaces, [:nif])
+            %{config: config, name: name, type: type, app: project.app}
+
+          _ ->
+            %{config: config, name: name, type: type, app: project.app}
+        end
```

Comment:
With this approach, you still have natives in either nifs, natives or cnodes and need to care about that everywhere. Instead, convert the input to the new format and supply all the defaults at the very beginning. This will also allow getting rid of the `:type` key in `Bundlex.Native`.

Reply:
Possibly this can be done in `Bundlex.Project.get`, so that Unifex didn't have to support deprecated format too

Reply:
I am just not sure if we can get rid of type because it is used in gcc.ex to indicate flags that should be appended and this probably cannot be done relying only on native interface but I will see :+1: 

https://github.com/membraneframework/bundlex/blob/2bd83f0e5a393f1c91c89bbe65a7ddb844eaff9f/lib/bundlex/toolchain/gcc.ex#L23-L25

Reply:
Fixed this, and moved to `Bundlex.Project.get` but could not remove `:type` key